### PR TITLE
Improve various tiny things.

### PR DESCRIPTION
**Pull request by ml-**
### Changes

- Remove variable in install= to shut up the vim lint. It's fixed anyway
- arch: Rename armv7 to armv7h
- Replace fixed _commit with magnificent and mostly unknown magic trick.
- yarn: Drop --non-interactive. I assume it's assumed when invoked by makepkg
- Move `date` inside of -ldflags. Cmd is short enough and good to read.
- Do the opposite of what I preached earlier and add CGO_* vars (Sorry)
- Add (incomplete) GOFLAGS env. It is exported, therefore picked up by
  `go test` as well, but without -trimpath.

### Notes/Questions

Sorry about changing my mind and adding all CGO_*, Foxboron hinted on irc that they might all be useful. Nothing to lose.

So apparently `armv7` is wrong and it's actually `armv7h`. Pretty sure I made that mistake on most of my other Go packages as well.
Are other ARM architectures supported as well?

### Commit ID/SHA

So I pretty much saw it coming that one day I'll forget to update the `_commit` variable on a version-bump. I quickly rejected the idea of cloning the whole git tag just to get the commit sha that I can pass to the build command. It would take much more time and space for almost zero gain.

For a reason unfathomable to me I raw hexdump on the tar from Github and surprisingly saw the commit id in the header. Turns out `git archive` puts the commit id in the tar file, which is quite nice. So this pretty much works on every Git repository site that uses `git archive` when creating releases (Github, Gitlab and probably most if not all other). The (very smart) Git developers also created a convenient command to read it `git get-tar-commit-id`.

I'm suprised this is not used more often. Most likely packagers just don't know about it. But enough of personal stories. Thanks for reading though.

This adds `gzip` to makedepends because of `zcat`. Which is most likely installed anyway.

### CODE REVIEW

File: PKGBUILD [hunk 1]
```diff
@@ -29,23 +28,25 @@ prepare() {
 }
 
 build() {
+  local _commit=$(zcat "${pkgname}-${pkgver}.tar.gz" | git get-tar-commit-id)
```
Comment (jmattheis) [inline PKGBUILD:31]: Nice!

**Review by jmattheis — APPROVED** [review]
Thanks for the great explanation :D.

File: PKGBUILD (hunk 1 above)
Comment (ml-) [reply PKGBUILD:31]: That's the closest I got to real magic.

**Comment by ml-** [comment]
Sorry, totally forgot. It's better if the reviewer merges I guess.

**Comment by jmattheis** [comment]
I prefer when the author merges, but it is probably personal preference.

**Comment by ml-** [comment]
Alright, but the reviewer should poke the author if it takes too long.

Also: want to push to AUR?

**Comment by jmattheis** [comment]
Alright, seems good. Yeah, I'll push it.

**Pull request merged**